### ISSUE TITLE
Advanced Logging example throws Null Reference exceptions

### ISSUE THREAD

**Issue by SOE-YoungS** [Azure-Samples/openai-python-enterprise-logging#16]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

Lines 42 & 65 of the Advanced Logging policy example [here](https://github.com/Azure-Samples/openai-python-enterprise-logging/blob/main/assets/apim-policy-event-hub-logging.xml), cause null reference exceptions. 

### Minimal steps to reproduce
>
Deploy the resources required for the Advanced Logging.

Copy the example policy, minus lines 19 through 29 (I did not need this level of validation, as we're using APIM Subscriptions in place of direct access via RBAC.).  

### Any log messages given by the failure
>
![image](https://github.com/Azure-Samples/openai-python-enterprise-logging/assets/95053834/a9fb677d-416e-4e39-9c05-1727c5dc9486)

### Expected/desired behavior
>
A value would be retrieved for the AppId.

### Mention any other details that might be useful
Removing these line solves the issue, but the AppId needs to be retrieved another way.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by SOE-YoungS** [comment]
To whoever picks up this issue:

Embeddings are not currently covered in the example (the main reason I looked at it).

See the below code for working implementation that allows for counting of tokens for Embeddings as well as chat completions / completions.

``` 
<log-to-eventhub logger-id="Request-Logger" partition-id="1">@{

                var responseBody = context.Response.Body?.As<JObject>(true);
				var operation = context.Operation.Id;
                string response = responseBody["choices"]?.ToString();
				
                string result = string.Empty;

				switch(operation)
				{
					case "ChatCompletions_Create":
					case "Completions_Create":	
						
						result = new JObject(
							new JProperty("event-time", DateTime.UtcNow.ToString()),
							new JProperty("backend", context.Request.Url.Host.ToString()),
							new JProperty("message-id", context.Variables["message-id"]),
							new JProperty("choices",response),
							new JProperty("operation", responseBody["object"].ToString()),
							new JProperty("model", responseBody["model"].ToString()),
							new JProperty("modeltime", context.Response.Headers.GetValueOrDefault("Openai-Processing-Ms",string.Empty)),
							new JProperty("completion_tokens", responseBody["usage"]["completion_tokens"].ToString()),
							new JProperty("prompt_tokens", responseBody["usage"]["prompt_tokens"].ToString()),
							new JProperty("total_tokens", responseBody["usage"]["total_tokens"].ToString()),
						).ToString();
						break;
					
					case "embeddings_create":
						result = new JObject(
							new JProperty("event-time", DateTime.UtcNow.ToString()),
							new JProperty("backend", context.Request.Url.Host.ToString()),
							new JProperty("message-id", context.Variables["message-id"]),
							new JProperty("operation", responseBody["object"].ToString()),
							new JProperty("model", responseBody["model"].ToString()),
							new JProperty("modeltime", context.Response.Headers.GetValueOrDefault("Openai-Processing-Ms",string.Empty)),
							new JProperty("prompt_tokens", responseBody["usage"]["prompt_tokens"].ToString()),
							new JProperty("total_tokens", responseBody["usage"]["total_tokens"].ToString()),
						).ToString();
						break;
				}
                return result;
            }</log-to-eventhub>

**Comment by SOE-YoungS** [comment]
This Null Ref issue was due to the requests being made (embedding generation requests) not containing a "messages" item in the response body.

I've added an updated policy example that covers all existing operations in [PR #20](https://github.com/Azure-Samples/openai-python-enterprise-logging/pull/20)

**Comment by SOE-YoungS** [comment]
Closed: [#20]
